packages/java/endpoint/src/test/java/com/vaadin/hilla/crud/filter/FilterTransformerTest.java

```diff
@@ -89,4 +90,25 @@ public void testRemap() {
         assertEquals(4, result2.getIntValue());
         assertEquals(TestEnum.TEST3, result2.getEnumValue());
     }
+
+    @Test
+    public void testUUIDFilter() {
+        var uuidString = "a012d576-4e96-40a0-9de0-67adb8f47023";
+
+        var testObject = new TestObject();
+        testObject.setName("test1");
+        testObject.setUuidValue(UUID.fromString(uuidString));
+        entityManager.persist(testObject);
+
+        var filter = new PropertyStringFilter("uuidValue", Matcher.EQUALS,
+                uuidString);
+
+        var spec = JpaFilterConverter.toSpec(filter, TestObject.class);
+        var result = repository.findAll(spec);
+        assertEquals(1, result.size());
+
+        var result1 = result.get(0);
+        assertEquals("test1", result1.getName());
+        assertEquals(UUID.fromString(uuidString), result1.getUuidValue());
+    }
```

Comment:
This test does not belong to `FilterTransformer`. It should go in `FilterTest`. You'll see that we have 4 tests for each type and you can mimic those tests for UUID.